src/utils/utilities.ts

```diff
@@ -271,20 +272,23 @@ export const generateDataFromPSBT = (base64Str: string, signer: Signer) => {
             masterFingerprint: derivation.masterFingerprint.toString('hex'),
             pubKey: derivation.pubkey.toString('hex'),
           };
+          signersList.push(data);
           if (data.masterFingerprint.toLowerCase() === signer.masterFingerprint.toLowerCase()) {
             // validating further by matching public key
             const xPub = signer.signerXpubs[XpubTypes.P2WSH][0].xpub; // to enable for taproot in future
-            const node = bip32.fromBase58(
-              xPub,
-              isTestnet() ? bitcoin.networks.testnet : bitcoin.networks.bitcoin
-            );
-            const receiveAddPath = data.derivationPath.split('/').slice(-2).join('/');
-            const childNode = node.derivePath(receiveAddPath);
-            const pubkey = childNode.publicKey.toString('hex');
+            const pubkey = getPubKeyFromXpub(xPub, data.derivationPath);
             if (data.pubKey == pubkey) signerMatched = true;
           }
         });
       }
+      const script = input.redeemScript || input.witnessScript;
```

Comment:
We only support P2WSH for remote signing for now so we should just check the witness script.

Also with Miniscript the script could include various check multisig opcodes. I think it's actually not necessary for us to check the m and n, and instead we could check that all the inputs addresses belong to/ can be derived from the wallet in question.

Reply:
For each input we can check the if it's change or not based on the last 2 parts of the derivation (which should be the same for all signers of the input) and check the address derived matches the address of the input